juju/juju
# application Update problematic for charmhub integration and too many ways to update config

## Description of change

Update in the application facade is problematic for the CharmHub integration has it allows another way a charm to be upgraded.  Update is only used by the juju client to update charm and application config.  Its full capabilities appear to be a duplicate of SetCharm. A full duplicate of which is no longer required. Changing to only be part of the application facade prior to version 13.  Setting a charm-url via Update is now unsupported.

We update charm and application config with 1 api call if given key value pairs on the cli, and a 2nd way when deploying bundles and when a file is provided on the cli, a third way when deploying charms or upgrading them.  Too confusing.  Consolidating the first 2 ways into 1 from here forward.   The prior methods are left for client compatibility with older versions of juju.

## QA steps

There should be no change to current behavior.  Test client with old versions of juju.  Test old clients with new juju controller.

Deploy bundles with config and use the juju config command.
